### PR TITLE
Helm preparation to get component versions from releases

### PR DESCRIPTION
**Pull request by tuommaki**
These changes prepare Helm deployment templates to get cluster component
versions from releases instead of k8scloudconfig.

Towards https://github.com/giantswarm/giantswarm/issues/8436

### CODE REVIEW

File: helm/azure-operator-chart/templates/rbac.yaml [hunk 1]
```diff
@@ -58,14 +64,6 @@ rules:
       - get
       - list
       - watch
-  - apiGroups:
-      - ""
-    resources:
-      - configmaps
-    resourceNames:
-      - aws-operator-configmap
```
Comment (tuommaki) [inline helm/azure-operator-chart/templates/rbac.yaml]: I'm pretty sure this wasn't needed :joy:

File: helm/azure-operator-chart/templates/deployment.yaml [hunk 2]
```diff
@@ -21,37 +21,37 @@ spec:
       annotations:
         releasetime: {{ $.Release.Time }}
       labels:
-        app: azure-operator
+        app: {{ tpl .Values.resource.default.name . }}
         version: {{ .Chart.Version }}
     spec:
       volumes:
-      - name: azure-operator-configmap
+      - name: {{ .Values.project.name }}-configmap
```
Comment (fiunchinho) [inline helm/azure-operator-chart/templates/deployment.yaml]: I had the same question when flattening: why do we sometimes use `.Values.project.name` and some other times `tpl .Values.resource.default.name`?

Comment (tuommaki) [reply helm/azure-operator-chart/templates/deployment.yaml]: I asked about this this in the `aws-operator` PR and was thinking the reason is probably that you want to have multiple operators running in parallel so resource names are distinct per release/version, but the YAML content for manifests are still the same across different versions.

Comment (kopiczko) [reply helm/azure-operator-chart/templates/deployment.yaml]: Stuff in `values.yaml` is not templated. So if you used `{{ .Values.resource.default.name . }}` the output would be `{{ .Release.Name | replace "." "-" | trunc 47 }}`. with `tpl` it actually templates the value output before rendering.

Comment (kopiczko) [reply helm/azure-operator-chart/templates/deployment.yaml]: I misinterpreted the original question. I talked with @fiunchinho and the outcome is we shouldn't use `.Values.project.name` in general.

**Comment by tuommaki** [comment]
Deploying this works now.

**Pull request merged**